bin/remove_empty_rows_and_columns.py

```diff
@@ -0,0 +1,83 @@
+"""
+Remove empty columns and rows from all CSV files in a directory.
+
+This is a program for use in development. It modifies source files by
+removing any columns with an empty heading and any entirely empty rows.
+"""
+
+import glob
+import os
+import logging
+import csv
+from argparse import ArgumentParser
+
+VERSION = '0.0.1'
+
+
+def removeIndices(list, indiciesToRemove):
+    for index in sorted(indiciesToRemove, reverse=True):
+        if len(list) > index:
+            del list[index]
+
+
+def main():
+    """Remove empty columns and rows from CSV files in a directory."""
+    parser = ArgumentParser(description='Remove empty columns and rows from CSV files in a directory.',
+                            epilog=f'Version: {VERSION}')
+
+    parser.add_argument('-i', '--input-dir',
+                        type=str,
+                        required=True,
+                        help='Input directory containing CSV files')
+
+    parser.add_argument('-o', '--output-dir',
+                        type=str,
+                        required=True,
+                        help='Output directory to write modified files')
+
+    args = parser.parse_args()
+
+    logging.basicConfig(
+        format='t=%(asctime)s lvl=%(levelname)s msg=%(message)s', level='INFO')
+
+    for directory in (args.input_dir, args.output_dir):
+        if not os.path.isdir(directory):
+            raise ValueError(
+                f'{directory} does not exist or is not a directory')
+
+    for filename in glob.glob(os.path.join(args.input_dir, '*.csv')):
+        basename = os.path.basename(filename)
+        out_filename = os.path.join(args.output_dir, basename)
+        with open(out_filename, 'w', encoding='utf-8-sig') as outfile:
+            with open(filename, newline='', encoding='utf-8-sig') as infile:
+                reader = csv.reader(infile)
+                headings = next(reader)
+                indicesToRemove = []
+                i = 0
+                for col in headings:
+                    if col == '':
```

Comment:
You should probably trim whitespace before checking the heading is empty.

Reply:
Agreed, and doing this for data cells too.